### PR TITLE
Issue #9007: Create jdk15 Input files compilation job in Travis

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #9007: Create jdk15 Input files compilation job in Travis

Since Travis is loaded with many jobs, I have decided to use semaphore for this execution.  I wasn't sure if I should make the changes in `InputRequireThisRecordCompactCtors.java` a separate commit or not, since I am using it as proof that this CI execution works as intended.

### CODE REVIEW

**Comment by nmancus1** [comment]
@romani In order to use jdk15 in semaphore, we will need to install it via PPA or wget.  Is this a problem?  If I had to guess, this will be the case for most CI regardless of what we use, since most official repositories will not have openjdk15 package builds yet.

**Comment by romani** [comment]
Github, rebase

**Comment by romani** [comment]
We can use any CI that supports jdk15.
We can install it as you suggested, but probably it will take awhile , so better to use CIs ready to use installation

**Review by romani — APPROVED** [review]
if CI is happy, I am ok to merge.

**Comment by nmancus1** [comment]
> We can install it as you suggested, but probably it will take awhile

Most of the other Linux CI use older Linux distros (travis uses Ubuntu 16.04.7 for example) than Semaphore(Ubuntu 18.04) does, so we will need to install it for those, too.  I think this is a good runtime, though:
image

It looked like a lot more work to do this in Cirrus, but I am not very familiar with Windows, either. If someone can recommend a better way to do this, I will implement it, but this seems like a good solution for now until jdk15 is in official repos.

File: .ci/validation.sh [hunk 1]
```diff
@@ -133,6 +133,20 @@ no-violation-test-josm)
   if [[ ${RESULT} != 0 ]]; then false; fi
   ;;
 
+javac15)
+  files=($(grep -Rl --include='*.java' ': Compilable with Java15' \
+        src/test/resources-noncompilable || true))
+  if [[  ${#files[@]} -eq 0 ]]; then
+    echo "No Java15 files to process"
+  else
+      mkdir -p target
```
Comment (rnveach) [inline .ci/validation.sh:142]: `mkdir` and `echo` don't have the same indentation.

Comment (nmancus1) [reply .ci/validation.sh]: Fixed.

**Comment by romani** [comment]
sorry, but in drone.io it is super easy task.
I merged fix.

**Pull request closed without merging**